app/models/status.rb

```diff
@@ -188,6 +188,8 @@ def permitted_for(target_account, account)
 
     def filter_timeline(query, account)
       blocked = Rails.cache.fetch("exclude_account_ids_for:#{account.id}") { Block.where(account: account).pluck(:target_account_id) + Block.where(target_account: account).pluck(:account_id) + Mute.where(account: account).pluck(:target_account_id) }
+      domains = Rails.cache.fetch("exclude_domains_for:#{account_id}") { AccountDomainBlock.where(account: account).pluck(:domain) }
```

Comment:
Instead of a local variable here, how about a (Rails cached) method like `Account#blocked_domains`? which the next line could call directly